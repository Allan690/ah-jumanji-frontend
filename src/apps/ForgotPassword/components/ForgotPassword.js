/* eslint-disable react/no-unescaped-entities */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/no-unknown-property */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';

class ForgotPassword extends Component {
  render() {
    return (
      <div className="row login-container">
        <div className="col-md-6 offset-md-3">
          <div className="bg-circle-1 bg-circle" />
          <div className="bg-circle-2 bg-circle" />
          <div className="bg-circle-3 bg-circle" />
          <div className="panel panel-default" id="login-form">
            <div className="panel-heading">
              <h3>Forgot Password.</h3>
              <hr />
            </div>
            <div className="panel-body">
              <form>
                <div className="form-group">
                  <label htmlFor="email">Email address</label>
                  <input type="email" className="form-control" id="email" placeholder="Enter email" required />
                </div>
                <button type="submit" className="btn btn-success btn-block">Submit</button>
                <hr />
                Don't have an account?
                <a href="/">Register.</a>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default ForgotPassword;
