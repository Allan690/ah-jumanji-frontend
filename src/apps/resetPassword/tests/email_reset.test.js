import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from "redux-thunk";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {MemoryRouter} from 'react-router-dom';
import ForgotPassword from '../components/PasswordResetComponent';
import ResetCode from "../components/PasswordResetCodeComponent";
import EmailReset from "../components/emailForm";
import CodeCollectionForm from "../components/collectResetURLForm";


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

Enzyme.configure({adapter: new Adapter()});

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, {context});
};

const mountWithStore = (component, store) => {
  return mount(
    <Provider store={store}>
      <MemoryRouter>
        {component}
      </MemoryRouter>
    </Provider>
  );
};


describe('ForgotPasswordComponent', () => {
  it('should render  ', () => {
    const component = shallow(<ForgotPassword debug />);

    expect(component).toMatchSnapshot();
  });

  it("should render an email field", () => {
    const component = shallow(<ForgotPassword debug />);

    expect(component.find('.form-control')).toBeDefined();
  });

  it("should render a submit button", () => {
    const component = shallow(<ForgotPassword debug />);

    expect(component.find('.form-btn')).toBeDefined();
  });

  it("should render an email component", () => {
    const component = shallow(<ForgotPassword debug />);

    expect(component.find('EmailReset')).toBeDefined();

  });

});

describe('Collect password updates', () => {
  let reset_code = {
    params: {
      reset_code: ""
    }
  };

  const component = shallow(<ResetCode debug match={reset_code} />);

  it('should render ', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have a notifications component', () => {
    expect(component.find('Notifications')).toBeDefined();
  });
});

describe("Render reset email component", () => {

  const InitialState = {
    resetAccountPassword: {}
  };
  const store = mockStore(InitialState);


  it('should render the form', function () {
    let component = shallowWithStore(<EmailReset />, store);
    expect(component).toMatchSnapshot();
  });

  it('should render self and subcomponents', () => {
    let component = mountWithStore(<EmailReset />, store);

    expect(component.find('button')).toBeDefined();
    expect(component.find('form')).toBeDefined();
    expect(component.find('Header')).toBeDefined();
    component = component.find('form').simulate('submit');
  });

  it('should reject invalid email', function () {
    let component = mountWithStore(<EmailReset />, store);

    let input = component.find('#email');
    expect(input).toBeDefined();
    input.simulate('change', {target: {value: 'Hello'}});
  });

});

describe("Render code collection form ", () => {
  const component = shallow(<CodeCollectionForm />);
  const InitialState = {
    resetAccountPassword: {}
  };
  const store = mockStore(InitialState);

  it('should render the form', function () {
    expect(component).toMatchSnapshot();
  });

  it('should have props undefined', function () {
    expect(component.props().recovery).toEqual(undefined);
  });

  it('should have a clickable button', function () {

    let component = mountWithStore(<CodeCollectionForm />, store);

    expect(component.find('button')).toBeDefined();
    expect(component.find('form')).toBeDefined();
    expect(component.find('Header')).toBeDefined();
    component.find('form').simulate('submit');

  });

  it('should reject non matching input', function () {
    let component = mountWithStore(<CodeCollectionForm />, store);
    component.debug();
    let input = component.find('#password1');
    let input2 = component.find('#password2');
    expect(input).toBeDefined();
    expect(input2).toBeDefined();
    input.simulate('change', {target: {value: 'Hello'}});
    input2.simulate('change', {target: {value: 'not HEllo'}});
    component.find('form').simulate('submit');


  });
});

