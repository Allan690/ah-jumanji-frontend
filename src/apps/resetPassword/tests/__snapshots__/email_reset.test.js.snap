// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collect password updates should render  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetCode
    debug={true}
    match={
      Object {
        "params": Object {
          "reset_code": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-md-6 offset-md-3"
      >
        <div
          className="bg-circle-1 bg-circle"
        />
        <div
          className="bg-circle-2 bg-circle"
        />
        <div
          className="bg-circle-3 bg-circle"
        />
        <Connect(withRouter(CodeCollectionForm))
          reset_code=""
        />
      </div>,
      "className": "row login-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bg-circle-1 bg-circle"
          />,
          <div
            className="bg-circle-2 bg-circle"
          />,
          <div
            className="bg-circle-3 bg-circle"
          />,
          <Connect(withRouter(CodeCollectionForm))
            reset_code=""
          />,
        ],
        "className": "col-md-6 offset-md-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-circle-1 bg-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-circle-2 bg-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-circle-3 bg-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "reset_code": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md-6 offset-md-3"
        >
          <div
            className="bg-circle-1 bg-circle"
          />
          <div
            className="bg-circle-2 bg-circle"
          />
          <div
            className="bg-circle-3 bg-circle"
          />
          <Connect(withRouter(CodeCollectionForm))
            reset_code=""
          />
        </div>,
        "className": "row login-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="bg-circle-1 bg-circle"
            />,
            <div
              className="bg-circle-2 bg-circle"
            />,
            <div
              className="bg-circle-3 bg-circle"
            />,
            <Connect(withRouter(CodeCollectionForm))
              reset_code=""
            />,
          ],
          "className": "col-md-6 offset-md-3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-circle-1 bg-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-circle-2 bg-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-circle-3 bg-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "reset_code": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ForgotPasswordComponent should render   1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-md-6 offset-md-3"
      >
        <div
          className="bg-circle-1 bg-circle"
        />
        <div
          className="bg-circle-2 bg-circle"
        />
        <div
          className="bg-circle-3 bg-circle"
        />
        <Connect(withRouter(EmailReset)) />
      </div>,
      "className": "row login-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bg-circle-1 bg-circle"
          />,
          <div
            className="bg-circle-2 bg-circle"
          />,
          <div
            className="bg-circle-3 bg-circle"
          />,
          <Connect(withRouter(EmailReset)) />,
        ],
        "className": "col-md-6 offset-md-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-circle-1 bg-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-circle-2 bg-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-circle-3 bg-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md-6 offset-md-3"
        >
          <div
            className="bg-circle-1 bg-circle"
          />
          <div
            className="bg-circle-2 bg-circle"
          />
          <div
            className="bg-circle-3 bg-circle"
          />
          <Connect(withRouter(EmailReset)) />
        </div>,
        "className": "row login-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="bg-circle-1 bg-circle"
            />,
            <div
              className="bg-circle-2 bg-circle"
            />,
            <div
              className="bg-circle-3 bg-circle"
            />,
            <Connect(withRouter(EmailReset)) />,
          ],
          "className": "col-md-6 offset-md-3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-circle-1 bg-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-circle-2 bg-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-circle-3 bg-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render code collection form  should render the form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(withRouter(CodeCollectionForm)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": Object {},
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render reset email component should render the form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(withRouter(EmailReset)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
